### PR TITLE
[CI] On-board the build pipeline to 1ES.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Few comments:

1. Extends cannot be used on templates. So we have to do a single extend and have duplicated code.
2. There are some common templates that we are working around using the use1ES parameter.
3. We are reusing the configure steps on other pipelines. That step should only be done in the build, that change is too big for this PR.
4. The governance template is not longer needed since the 1ES template provides it.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: tools/devops/automation/build-pipeline.yml [hunk 1]
```diff
@@ -27,10 +211,60 @@ trigger:
...
 extends:
-  template: templates/pipelines/build-pipeline.yml
+  template: azure-pipelines/MicroBuild.1ES.Official.yml@CustomPipelineTemplates
   parameters:
-    isPR: false
-    signingSetupSteps: 
-      - template: ./templates/sign-and-notarized/setup.yml
-        parameters:
-          isPR: false
+    pool:  # default pool to be used forvalidation jobs
```
Comment (rolfbjarne) [inline tools/devops/automation/build-pipeline.yml:216]:
```suggestion
    pool:  # default pool to be used for validation jobs
```

File: tools/devops/automation/build-pipeline.yml [hunk 2]
```diff
@@ -27,10 +211,60 @@ trigger:
...
+      name: AzurePipelines-EO
+      image: 1ESPT-Windows2022
+      os: windows
+    sdl:
+      baseline:
+        baselineFile: '$(System.DefaultWorkingDirectory)\\maccore\\tools\\devops\\baselines.gdnbaselines'
+      suppression:
+        suppressionFile: '$(System.DefaultWorkingDirectory)\\maccore\\tools\\devops\\suppress.gdnsuppress'
```
Comment (rolfbjarne) [inline tools/devops/automation/build-pipeline.yml:224]: Can we put these files in xamarin-macios instead? To reduce the dependency on maccore...

File: tools/devops/automation/templates/common/mac-agent-logs.yml [hunk 3]
```diff
@@ -49,14 +53,26 @@ steps:
...
+      displayName: 'Publish Artifact: Agent logs'
+      condition: always()
+      inputs:
+        # add or not the working dir to the upload
+        ${{ if eq(parameters.workingDirectory, '') }}:
+          targetPath: ${{ parameters.outputPath }}
+        ${{ else }}:
+          targetPath: ${{ parameters.workingDirectory }}/${{ parameters.outputPath }} 
+        artifactName: '${{ parameters.uploadPrefix }}$(Agent.Name)-$(System.JobName)-$(System.JobId)-${{ parameters.outputPath }}'
+      continueOnError: true
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/common/mac-agent-logs.yml:78]: It doesn't work to simplify the condition to have only the single different line inside the if?

```suggestion
  - ${{ if eq(parameters.use1ES, true) }}:
    - task: 1ES.PublishPipelineArtifact@1
  - ${{ else }}:
    - task: PublishPipelineArtifact@1
    displayName: 'Publish Artifact: Agent logs'
    condition: always()
    inputs:
      ${{ if eq(parameters.workingDirectory, '') }}:
        path: ${{ parameters.outputPath }}
      ${{ else }}:
        path: ${{ parameters.workingDirectory }}/${{ parameters.outputPath }} 
      artifact: '${{ parameters.uploadPrefix }}$(Agent.Name)-$(System.JobName)-$(System.JobId)-${{ parameters.outputPath }}'
    continueOnError: true
```

File: tools/devops/automation/templates/main-stage.yml [hunk 4]
```diff
@@ -302,11 +305,13 @@ stages:
   ${{ if and(ne(parameters.dependsOn, ''), ne(parameters.dependsOnResult, '')) }}:
     condition: eq(dependencies.${{ parameters.dependsOn }}.result, '${{ parameters.dependsOnResult }}')
   jobs:
-
+
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/main-stage.yml:308]:
```suggestion

```

File: tools/devops/automation/templates/sign-and-notarized/funnel.yml [hunk 5]
```diff
@@ -144,6 +145,17 @@ jobs:
 - job: upload_azure_blob
   displayName: 'Upload packages to Azure & SBOM'
   timeoutInMinutes: 1000
+  templateContext: # disable some of the sdl tasks for this kjob
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/sign-and-notarized/funnel.yml:148]:
```suggestion
  templateContext: # disable some of the sdl tasks for this job
```

File: tools/devops/automation/templates/sign-and-notarized/funnel.yml [hunk 6]
```diff
@@ -144,6 +145,17 @@ jobs:
...
+    sdl:
+      credscan: # run in the sdl stage 
+        suppressionsFile: '$(System.DefaultWorkingDirectory)\\maccore\\tools\\devops\\CredScanSuppressions.json'
+        enabled: false
+      binskim: # job does not compile anything
+        enabled: false
+      eslint: # no js or anything similar
+        enabled: false
+      bandit: # we have no pyhon
```
Comment (rolfbjarne) [inline tools/devops/automation/templates/sign-and-notarized/funnel.yml:157]:
```suggestion
      bandit: # we have no python
```

**Review by rolfbjarne — APPROVED** [review]

File: tools/devops/automation/build-pipeline.yml (hunk 2 above)
Comment (mandel-macaque) [reply tools/devops/automation/build-pipeline.yml:224]: Will do that in a following PR.

File: tools/devops/automation/templates/common/mac-agent-logs.yml (hunk 3 above)
Comment (mandel-macaque) [reply tools/devops/automation/templates/common/mac-agent-logs.yml:78]: I wish, that would work if the 1ES people chose to make the task parameters compatible with the default task. But they decided to do:

* targetPath -> path

Artefact does work AFAIK: https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-pipeline-artifact-v1?view=azure-pipelines but we would make it worse due to targetPath.

**Review by haritha-mohan — APPROVED** [review]

**Pull request merged**